core/src/main/java/org/apache/calcite/sql/type/SqlTypeFactoryImpl.java

```diff
@@ -88,6 +90,17 @@ public SqlTypeFactoryImpl(RelDataTypeSystem typeSystem) {
     if (maxPrecision >= 0 && precision > maxPrecision) {
       precision = maxPrecision;
     }
+    if (typeName == SqlTypeName.DECIMAL) {
+      if (precision != RelDataType.PRECISION_NOT_SPECIFIED && precision <= 0) {
+        throw RESOURCE.invalidPrecisionForDecimalType(precision, maxPrecision).ex();
+      }
+      if (scale != RelDataType.SCALE_NOT_SPECIFIED && scale < 0) {
+        if (!typeSystem.supportsNegativeScale() || scale < typeSystem.getMinNumericScale()) {
```

Comment:
is the first test necessary?
comparing the scale with the minimum one may be enough?

Reply:
I will remove it.